### PR TITLE
New digiposte konnector

### PR DESCRIPTION
**Pull request by doubleface**
Here is a new digiposte konnector, which only gets bills at the moment but it's not too bad I suppose.

Fixes also #182

### CODE REVIEW

File: build/server/konnectors/digiposte.js [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+    connector.logger.info('Number of fetched documents : ' + nb);
+    if (nb > 0) {
+      var localizationKey = 'notification bills';
+      var options = {
+        smart_count: nb
+      };
+      next(null, localization.t(localizationKey, options));
+    }
+  }).catch(function (err) {
+    next(err);
```
Comment (nicofrand) [inline build/server/konnectors/digiposte.js]: That would be better to log the error and return the actual locale key corresponding to the error. Ex: 'bad credentials'.

Comment (doubleface) [inline build/server/konnectors/digiposte.js]: This should be fixed now

Comment (nicofrand) [inline build/server/konnectors/digiposte.js]: There should be a parsing error here too : https://github.com/doubleface/konnectors/blob/e4228f3f0d3fa1542910a420a4902e554347ed60/server/konnectors/digiposte.js#L57 and here https://github.com/cozy-labs/konnectors/pull/724/files/6dfb55e3565604a6c473b5dedb24ca8b517d958b#diff-dab0f39cee0d1813f8adc9f00c8b9b78R109

Comment (doubleface) [inline build/server/konnectors/digiposte.js]: Wow fast review. Thank you!

This should be fixed now

Comment (nicofrand) [inline build/server/konnectors/digiposte.js]: BTW, we usually don't commit the build directory directly, is there a change inside cozy team ?

File: build/server/konnectors/digiposte.js [hunk 2]
```diff
@@ -108,7 +111,8 @@ function fetchBills(requiredFields, bills, data, next) {
       connector.logger.info('App access token is ' + accessToken);
       return accessToken;
     } else {
-      return Promise.reject('Problem fetching the access token');
+      connector.logger.error('Problem fetching the access token');
+      return Promise.reject('parsing error');
```
Comment (nicofrand) [inline build/server/konnectors/digiposte.js]: There is a dedicated locale key for that : https://github.com/cozy-labs/konnectors/blob/master/client/app/locales/en.json#L12

File: server/models/file.coffee [hunk 3]
```diff
@@ -93,10 +94,17 @@ File.createNew = (fileName, path, url, tags, callback) ->
...
 
     # Save file in a tmp folder while attachBinary supports stream.
-    options =
-        uri: url
-        method: 'GET'
-        jar: true
+    options = {}
+    if requestoptions
+        options = requestoptions
+        if _.isFunction options.form
```
Comment (ZeHiro) [inline server/models/file.coffee]: Why not using 'typeof options.form === ' function' instead ?

Comment (ZeHiro) [inline server/models/file.coffee]: BTW, love the feature :)

File: build/server/konnectors/digiposte.js [hunk 4]
```diff
@@ -0,0 +1,243 @@
+'use strict';
```
Comment (ZeHiro) [inline build/server/konnectors/digiposte.js]: Con you not push a built version of the K ? instead the rules have changed : - )

File: server/konnectors/digiposte.js [hunk 5]
```diff
@@ -0,0 +1,249 @@
...
+  },
+  dataType: ['bill'],
+  models: [Bill],
+  fetchOperations: [fetchBills]
+})
+
+function fetchBills (requiredFields, bills, data, next) {
+  let request = require('request-promise')
+  const j = request.jar()
+  // require('request-debug')(request)
```
Comment (ZeHiro) [inline server/konnectors/digiposte.js]: Forgotten comment

File: build/server/konnectors/digiposte.js (hunk 4 above)
Comment (doubleface) [inline build/server/konnectors/digiposte.js]: Ok, but should I push a built version of file and save_data_and_files ?

Comment (doubleface) [inline build/server/konnectors/digiposte.js]: I let you finish the review and will do a new pullrequest without the built files. Please don't merge it.

File: server/models/file.coffee (hunk 3 above)
Comment (doubleface) [inline server/models/file.coffee]: Juste because lodash is already here and it feels more explicit

File: server/konnectors/digiposte.js (hunk 5 above)
Comment (doubleface) [inline server/konnectors/digiposte.js]: That was on purpose, actually. It is here to make the debugging easier.

File: server/models/file.coffee (hunk 3 above)
Comment (nicofrand) [inline server/models/file.coffee]: or just `instanceof Function` :p ?

Comment (doubleface) [inline server/models/file.coffee]: Ok, ok :)

File: server/konnectors/digiposte.js (hunk 5 above)
Comment (doubleface) [inline server/konnectors/digiposte.js]: And this module is cool for developing new konnectors

**Comment by doubleface** [comment]
Made a new pullrequest without the build files : #725

**Pull request closed without merging**